src/Makefile

```diff
@@ -1,7 +1,38 @@
+-include ../config.mk
 
-CC = gcc
-LUAJIT_FLAGS = `pkg-config --cflags --libs luajit`
+PREFIX ?= /usr/local
+BINDIR ?= $(PREFIX)/bin
+LIBDIR ?= $(PREFIX)/lib
+CC ?= gcc
 
-main: main.c
-	$(CC) -o main main.c $(LUAJIT_FLAGS) -Wl,--no-as-needed -Wl,-E
+LUA_SRCS := \
+	entrypoint.lua \
+	main.lua \
+	lib/glib.lua \
+	lib/wpe.lua \
+	apps/singleview/main.lua
 
+LUA_OBJS := \
+	$(patsubst %.lua,%_lua.o,$(LUA_SRCS))
+
+CFLAGS += $(shell pkg-config luajit --cflags)
```

Comment:
“Normal” assignment (`=`) in Make is lazy, and the shell will be invoked every time that `$(CFLAGS)` and `$(LDFLAGS)` are expanded. It would be better to pick first the outputs from `pkg-config` using immediate assignment (`:=`), and then add the result to the variables:

```make
# These are evaluated only once when the variables are assigned.
LUAJIT_CFLAGS := $(pkg-config luajit --cflags)
LUAJIT_LDLIBS := $(pkg-config luajit --libs)

# These are evaluated every time they are referenced.
CFLAGS  += $(LUAJIT_CFLAGS)
LDFLAGS += $(LUAJIT_LDLIBS)
```